augur/io/sequences.py

```diff
@@ -448,29 +448,41 @@ def _read_genbank(reference, feature_names):
     return features
 
 
-def validate_sequence_file(file: str):
-    """Ensure all sequence identifiers in the file are unique."""
-    with NamedTemporaryFile("w+", encoding="utf-8") as dup_num_file:
+def read_sequence_ids(file: str, error_on_duplicates = True):
```

Comment:
_non-blocking_

This is doing similar things to what I did with [`pyfastx` for augur curate](https://github.com/nextstrain/augur/blob/49197e07f43b98f1dad18fb36cc82dbf648957c0/augur/io/metadata.py#L379-L416). The main downside for `pyfastx` is that it can only handle plain and gzip FASTAs, but it's nice for random access of sequences. I think the use-cases are different enough that we don't need to consolidate the two but good to get other's thoughts here.